### DISCUSSION TITLE
Profile Templates in VSCodium?

### DISCUSSION THREAD

**Discussion by Carsten-R** [VSCodium/vscodium#1505] [Q&A]
1.78 introduced **Profile Templates** in VS Code. Is this feature proprietary to VS Code?
Is there a way to activate it in VSCodium?

**Comment by gbakeman** [comment]
Potentially related: https://github.com/VSCodium/vscodium/issues/1511 Closed as "not planned" in November 2023.

Looking in VSCodium 1.89.1, we do have the ability to Import and Export profile templates:

![image](https://github.com/VSCodium/vscodium/assets/842942/3eba2b1b-1c92-4d0a-a464-3ef01891f76d)

We can also create profiles from others:

![image](https://github.com/VSCodium/vscodium/assets/842942/ea9f8856-3e7f-4dda-b7d7-15588af61ccc)

VSCode:

> [comes with a predefined set](https://code.visualstudio.com/docs/editor/profiles#_profile-templates) of profile templates that you can use to customize VS Code for your specific workflow. To create a new profile based on a template, select a Profile Template when going through the Create Profile flow.

It just looks like VSCodium doesn't include the default set. I'm having a hard time finding a URL I can use to import the templates as well.

**Comment by NezSez** [comment]
As noted in 10/28/2024 [Microsoft VSCode Profiles documentation](https://code.visualstudio.com/docs/editor/profiles#_why-are-templates-not-available-when-creating-a-new-profile):

> Profile templates are hosted externally by VS Code

No idea where MS pulls these from but you should be able to find the URL in their repo. They could be copied if open sourced as well by the codium devs and distributed (shouldn't use too much space). Sry, I don't have the time to track this down.

**Reply by daiyam** [reply]
They are defined in the `product.json` (`"profileTemplatesUrl": "https://az764295.vo.msecnd.net/profile-templates/profile-templates.json",`).

We don't include them because they haven't been verified, some must depends on MS proprietary extensions and maybe on unavailable extensions on OpenVSX.

We could store custom templates directly on the repo if somebody have some time to do a PR.

**Reply by NezSez** [reply]
Yes, that's why I posted that note from MS...wouldn't be proper to possibly enable telemetry by accessing the profile site at MS's repo.
Would you be able to setup the service for it to be available via URL mimicking MS, or store them as statically included? One would have to update them to stay in sync with MS ver, etc.

**Reply by daiyam** [reply]
> Would you be able to setup the service for it to be available via URL mimicking MS, or store them as statically included?

I think since they are only JSON files.

> One would have to update them to stay in sync with MS ver, etc.

Do we? The extensions won't be the same so why bother?

**Reply by NezSez** [reply]
Well that's true, although some will/have probably used the MS extension site (or whatever it's called), but there are profile settings other than the extensions that MS will change/add as well.

**Comment by NezSez** [comment]
Possible workaround:
If you have MS VSCode installed somewhere you can export each of the templates, or create new ones based on them and export them to a github gist or a file then import them in Codium.
If you launch the profile URL from a github gist, it will open in the web version and you may be able to import/export from it as well.
